scp/lib/time_table.dart

```diff
@@ -26,38 +25,42 @@ class TimeTableState extends State<TimeTable> {
   final double unitHeight = 80.0;
   double screenWidth, screenHeight;
 
+
   Future _fetchSectionData(BuildContext context) async {
+    TimeTableResources.setCourseNumber(); // Sets the course number of Maths and Physics for corresponding semester.
+    int currentMonth = DateTime.now().month; // If current month is greater than august, usual TT, else modify.
     SharedPreferences prefs = await SharedPreferences.getInstance();
     theorySection = prefs.getString('theory_section');
     practicalSection = prefs.getString('prac_section');
-    if ((theorySection.compareTo('Ar.') == 0) ||
-        (theorySection.compareTo('A') == 0) ||
-        (theorySection.compareTo('D') == 0) ||
-        (theorySection.compareTo('C') == 0) ||
-        (theorySection.compareTo('B') == 0)) {
-      sectionSequence = 'tp';
+    if((theorySection.compareTo('Ar.')==0)||(theorySection.compareTo('A')==0)||(theorySection.compareTo('D')==0)||(theorySection.compareTo('C')==0)||(theorySection.compareTo('B')==0))
+      {
+        sectionSequence = currentMonth>=8 ? 'tp' : 'pt';
+      }else{
+      sectionSequence = currentMonth>=8 ? 'pt' : 'tp';
+    }
+
+    if(currentMonth < 8){
+      theorySection = TimeTableResources.substituteTheorySection[theorySection];
+      practicalSection = TimeTableResources.substitutePracticalSection[practicalSection];
     }
-    print("Sequence" + sectionSequence);
-    print("Theory" + theorySection);
-    print("Practical" + practicalSection);
   }
 
-  _resetSections(BuildContext context) async {
-    SharedPreferences sharedPreferences = await SharedPreferences.getInstance();
+  _resetSections(BuildContext context) async{
+    SharedPreferences sharedPreferences=await SharedPreferences.getInstance();
     sharedPreferences.getKeys();
     sharedPreferences.remove('theory_section');
     sharedPreferences.remove('prac_section');
     sharedPreferences.setBool('show_timetable', false);
-    Navigator.of(context).pushNamedAndRemoveUntil(
-        Routes.rHomepage, (Route<dynamic> route) => false);
+    Navigator.of(context)
+        .pushNamedAndRemoveUntil('/homePage', (Route<dynamic> route) => false);
```

Comment:
The removed line here is the actual route in upstream

Reply:
Okay okay. have changed this.